libvips/foreign/svgload.c

```diff
@@ -473,22 +473,12 @@ G_DEFINE_TYPE( VipsForeignLoadSvgStream, vips_foreign_load_svg_stream,
 gboolean
 vips_foreign_load_svg_stream_is_a( VipsStreami *input )
 {
-	ssize_t n;
+	unsigned char *data;
 
-	if( vips_streami_rewind( input ) )
+	if( !(data = vips_streami_sniff( input, SVG_HEADER_SIZE )) )
```

Comment:
This fails if the SVG image is smaller than `SVG_HEADER_SIZE` (1000). Since `vips_streami_sniff` will sniff exactly the amount of needed bytes. See for example:
```bash
$ wget https://dev.w3.org/SVG/tools/svgweb/samples/svg-files/intertwingly.svg
$ stat --printf="%s" intertwingly.svg
962
$ cat intertwingly.svg | vips invert stdin x.jpg
VipsForeignLoad: stream is not in a known format
$ printf '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n' > x.svg; cat intertwingly.svg >> x.svg;
$ stat --printf="%s" x.svg
1017
$ cat x.svg | vips invert stdin x.jpg
$ vipsheader x.jpg
x.jpg: 100x100 uchar, 3 bands, srgb, jpegload
```

Reply:
Ah I wondered what was going on there. Let's change sniff then to return the number of bytes read.

Reply:
Fixed with https://github.com/libvips/libvips/pull/1443/commits/998b5eea4593d92adaf2141854d6a7a55e453f43, thanks!